Tribler/Test/Core/test_session.py

```diff
@@ -187,3 +188,28 @@ def verify_load_checkpoint_call():
         self.session.lm.load_checkpoint = verify_load_checkpoint_call
         self.session.load_checkpoint()
         self.assertTrue(self.load_checkpoint_called)
+
+
+class TestSessionWithLibTorrent(TestSessionAsServer):
+
+    def setUpPreSession(self):
+        super(TestSessionWithLibTorrent, self).setUpPreSession()
+        self.config.set_libtorrent(True)
+
+    @deferred(timeout=10)
+    def test_remove_torrent_id(self):
+        """
+        Test whether removing a torrent id works.
+        """
+        torrent_def = TorrentDef.load(TORRENT_UBUNTU_FILE)
+        dcfg = DownloadStartupConfig()
+        dcfg.set_dest_dir(self.getDestDir())
+
+        # @deferred does not play nice with inlineCallbacks and blocking
+        @blocking_call_on_reactor_thread
+        @inlineCallbacks
+        def wait_for_dl_start():
+            yield self.session.start_download_from_tdef(torrent_def, dcfg=dcfg, hidden=True)
+        wait_for_dl_start()
+
+        return self.session.remove_download_by_id(torrent_def.infohash)
```

Comment:
😊 Woops, `start_download_from_tdef` returns the `LibtorrentDownloadImpl`. This should've yielded on `download.get_handle()`. I can chain in a callback which forwards `handle.info_hash()` to `remove_download_by_id` 👍